File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        without_dup = set()
        
        for i in nums:
            if i in without_dup:
                return i
            without_dup.add(i)
        

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
    let dup = new Set();
    for (let i = 0; i < nums.length; i++) {
        if (dup.has(nums[i])) {  
            return nums[i];     
        }
        dup.add(nums[i]);        
    }
};
